pkg/TIPA.py

```
@@ -158,593 +185,527 @@ class MessageFunc():
+        try:
+            if site not in ["market", "wiki"]:
+                raise ValueError("Invalid site. Choose 'market' or 'wiki'.")
+
+            search_url = self.construct_search_url(site, search_text)
+            page = requests.get(search_url, timeout=10)
+            page.raise_for_status()  # Raises an HTTPError if the status is 4xx, 5xx
+
+            soup = BeautifulSoup(page.content, 'html.parser')
+            search_results = soup.find("div", {"class": "egMi0 kCrYT"})
+            if not search_results:
+                raise Exception("No search results found.")
```

Ruff (0.8.0)

Abstract `raise` to an inner function

(TRY301)